src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Resource.Designer.targets

```diff
@@ -0,0 +1,216 @@
+    <!--
+      Handle the case where ResolveProjectReferences is running.
+      `ResolveProjectReferences` runs after a project has build, but
+      during the referencing project build system.
+
+      Unfortunately it gets passed the referencing projects `IntermediateOutputPath`
+      value, this can cause it to look for files which do not exist.
+    -->
+    <_DesignerIntermediateOutputPath Condition=" '$(_DesignerIntermediateOutputPath)' == '' And Exists ('$(IntermediateOutputPath)$(_DesignerAssemblyName).dll') ">$(IntermediateOutputPath)</_DesignerIntermediateOutputPath>
+    <_DesignerIntermediateOutputPath Condition=" '$(_DesignerIntermediateOutputPath)' == '' And '$(_OuterIntermediateOutputPath)' != '' And Exists ('$(_OuterIntermediateOutputPath)$(_DesignerAssemblyName).dll') ">$(_OuterIntermediateOutputPath)</_DesignerIntermediateOutputPath>
+    <!-- If we have not already built an assembly in either location, prefer the
+    `_OuterIntermediateOutputPath` if possible over the `IntermediateOutputPath`.
+    This will mean only one copy of the assembly will be created. Rather than one
+    per TargetFramework.
+    -->
+    <_DesignerIntermediateOutputPath Condition=" '$(_DesignerIntermediateOutputPath)' == '' And '$(_OuterIntermediateOutputPath)' != '' ">$(_OuterIntermediateOutputPath)</_DesignerIntermediateOutputPath>
+    <_DesignerIntermediateOutputPath Condition=" '$(_DesignerIntermediateOutputPath)' == '' ">$(IntermediateOutputPath)</_DesignerIntermediateOutputPath>
+    <_GenerateResourceDesignerAssemblyOutput>$(_DesignerIntermediateOutputPath)$(_DesignerAssemblyName).dll</_GenerateResourceDesignerAssemblyOutput>
+    <_GenerateResourceDesignerClassFile Condition=" '$(Language)' == 'F#' ">$(_DesignerIntermediateOutputPath)_$(_DesignerAssemblyName).fs</_GenerateResourceDesignerClassFile>
+    <_GenerateResourceDesignerClassFile Condition=" '$(_GenerateResourceDesignerClassFile)' == '' ">$(_DesignerIntermediateOutputPath)_$(_DesignerAssemblyName).cs</_GenerateResourceDesignerClassFile>
+    <_GenerateResourceCaseMapFile>$(_DesignerIntermediateOutputPath)case_map.txt</_GenerateResourceCaseMapFile>
+  </PropertyGroup>
+  <Message Text="_OuterIntermediateOutputPath: $(_OuterIntermediateOutputPath)" />
+  <Message Text="IntermediateOutputPath: $(IntermediateOutputPath)" />
+</Target>
+
+<Target Name="_GenerateRtxt"
+    Condition="'$(AndroidUseDesignerAssembly)' == 'True' And '$(DesignTimeBuild)' == 'True' "
+    DependsOnTargets="_CreatePropertiesCache;_ResolveSdks;_ResolveAndroidTooling;_GetJavaPlatformJar;_GenerateAndroidResourceDir;_SetupDesignerProperties"
+    Inputs="$(_AndroidResFlagFile);@(_AndroidResourceDest);@(LibraryResourceDirectories->'%(StampFile)')"
+    Outputs="$(_DesignerIntermediateOutputPath)R.txt"
+  >
+    <!-- Generate an R.txt file using the Managed Parser -->
+    <GenerateRtxt
+      RTxtFile="$(_DesignerIntermediateOutputPath)R.txt"
```

Comment:
Convention: XML properties should be indented an "additional tab stop", which you did for the `<Target/>`.  This line and the following lines should have an additional two spaces `  ` before them.  We may *also* want to consider *sorting* the properties, unless the order is important for clarity & understanding.  (I'm not sure that the order *is* important; that's your call.)